lib/src/gestures/gestures.dart

```diff
@@ -706,7 +759,7 @@ abstract class MapGestureMixin extends State<FlutterMap>
 
   double _getZoomForScale(double startZoom, double scale) {
     var resultZoom =
-        scale == 1.0 ? startZoom : startZoom + math.log(scale) / math.ln2;
+        startZoom + ((scale > 0 ? 1 : -1) * (math.log(scale.abs()) / math.ln2));
```

Comment:
You could add practical bounds here or inside `mapState.fitZoomToBounds`, for instance `max(0,min(zoomResult, 25)`